### ISSUE TITLE
Allow serialization of an invocation list with more than one target?

### ISSUE THREAD

**Issue by asusralis** [rikimaru0345/Ceras#60]
Hello! I was curious on why Ceras can currently only serialize delegates with an invocation length of one. Would it be difficult or impossible to allow multiple?

Also, it would be great to put the name of the delegate (like you do in ThrowStatic and ThrowInstance) so you know what delegate it is!

**Comment by rikimaru0345** [comment]
It would be very easy to add.
The only reason why it isn't already so is because I didn't need it myself.

Do you need it? If you do, let me know and I'll add it. Shouldn't take long :)

**Comment by asusralis** [comment]
That would be awesome! If I understand correctly, this would allow events with multiple subscribers to be serialized?

**Comment by rikimaru0345** [comment]
Yes, indeed. Events would then round-trip through the serialization.

This will be (another) big breaking change. Fortunately I'm currently already rewriting many parts of Ceras  for version 5 (see #53 ). So since I'm already trying to merge in as many breaking changes into v5 as possible, adding this will be not a problem.

Once it's done, you might have to switch to the (for now still experimental) v5 branch though because it might take a little longer until all the changes in v5 are done.

I'm looking into adding this feature right now. So if everything goes well I might even commit it today (in the next 1-3 hours or so)

**Comment by asusralis** [comment]
That's so cool! I'm looking forward to it, thanks!

**Comment by rikimaru0345** [comment]
https://github.com/rikimaru0345/Ceras/commit/c032e1ae20565b64b1922369c511b6b00d466ac8 has all the needed changes. But there are some other things still preventing the branch from being usable.

I'll take a look at those later to see if I can make a semi-usable version for testing or if I need to fully complete the work on some other parts of #53 first.
